app/src/main/java/fr/neamar/kiss/DataHandler.java

```diff
@@ -431,6 +432,53 @@ public SearchProvider getSearchProvider() {
         return favorites;
     }
 
+    /**
+     * This method is used to set the specific position of an app in the fav array.
+     *
+     * @param context The mainActivity context
+     * @param id the app you want to set the position of
+     * @param position the new position of the fav
+     */
+    public void setFavoritePosition(MainActivity context, String id, int position) {
+        String favApps = PreferenceManager.getDefaultSharedPreferences(this.context).
+                getString("favorite-apps-list", "");
+        List<String> favAppsList = new ArrayList<>(Arrays.asList(favApps.split(";")));
+
+        int currentPos = favAppsList.indexOf(id);
+        if(currentPos == -1) {
+            Log.e(TAG, "Couldn't find id in favAppsList");
+            // TODO: Perhaps we should throw a better error here?
+            return;
+        }
+        // Clamp the position so we dont just extend past the end of the array.
+        position = Math.min(position, favAppsList.size() - 1);
+
+        favAppsList.remove(currentPos);
+        // Because we're removing ourselves from the array, positions may change, we should take that into account
+        favAppsList.add(currentPos > position ? position + 1 : position, id);
+        String newFavList = TextUtils.join(";", favAppsList);
+
+        PreferenceManager.getDefaultSharedPreferences(context).edit()
+                .putString("favorite-apps-list", newFavList + ";").apply();
+
+        context.onFavoriteChange();
+    }
+
+    /**
+     * Helper function to get the position of a favorite. Used mainly by the drag and drop system to know where to place the dropped app.
+     *
+     * @param context mainActivity context
+     * @param id the app you want to get the position of.
+     * @return
+     */
+    public int getFavoritePosition(MainActivity context, String id) {
+        String favApps = PreferenceManager.getDefaultSharedPreferences(this.context).
+                getString("favorite-apps-list", "");
+        List<String> favAppsList = new ArrayList<>(Arrays.asList(favApps.split(";")));
+
+        return favAppsList.indexOf(id);
```

Comment:
same comment here as above, you might have some weird use cases when users have an item in their favorits that they uninstalled since then, using getFavorites() should remove this problem

Reply:
As above